test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs

```diff
@@ -200,44 +220,63 @@ public async Task VerifySignaturesAsync_ExpiredCertificateAndTimestampWithTooLar
             var bcCertificate = ca.IssueCertificate(issueOptions);
 
             using (testServer.RegisterResponder(timestampService))
-            using (var certificate = new X509Certificate2(bcCertificate.GetEncoded()))
             using (var directory = TestDirectory.Create())
             {
-                certificate.PrivateKey = DotNetUtilities.ToRSA(keyPair.Private as RsaPrivateCrtKeyParameters);
+#if IS_DESKTOP
+                var privateKey = DotNetUtilities.ToRSA(keyPair.Private as RsaPrivateCrtKeyParameters);
+#else
+                var rsaParameters = DotNetUtilities.ToRSAParameters(keyPair.Private as RsaPrivateCrtKeyParameters);
 
-                var packageContext = new SimpleTestPackageContext();
-                var signedPackagePath = await SignedArchiveTestUtility.AuthorSignPackageAsync(
-                    certificate,
-                    packageContext,
-                    directory,
-                    timestampService.Url);
+                RSACryptoServiceProvider rsaCsp = new RSACryptoServiceProvider();
 
-                var waitDuration = (issueOptions.NotAfter - DateTimeOffset.UtcNow).Add(TimeSpan.FromSeconds(1));
+                rsaCsp.ImportParameters(rsaParameters);
 
-                // Wait for the certificate to expire.  Trust of the signature will require a valid timestamp.
-                if (waitDuration > TimeSpan.Zero)
+                var privateKey = rsaCsp;
+#endif
+
+#if IS_DESKTOP
+                using (var certificate = new X509Certificate2(bcCertificate.GetEncoded()))
                 {
-                    await Task.Delay(waitDuration);
-                }
+                    certificate.PrivateKey = privateKey;
+#else
+                using (var certificateTmp = new X509Certificate2(bcCertificate.GetEncoded()))
+                using (var certificate = RSACertificateExtensions.CopyWithPrivateKey(certificateTmp, privateKey))
+                {
+#endif
```

Comment:
I feel that you can simplify this PR and reduce code duplication significantly with smart use of utility methods.

For example:
```C#
internal static class CertificateUtilities
{
    internal static X509Certificate2 GetCertificateWithPrivateKey(X509Certificate bcCertificate, RSA privateKey)
    {
        RSA privateKey;

#if IS_DESKTOP
        privateKey = DotNetUtilities.ToRSA(keyPair.Private as RsaPrivateCrtKeyParameters);
#else
        RSAParameters rsaParameters = DotNetUtilities.ToRSAParameters(keyPair.Private as RsaPrivateCrtKeyParameters); 
        var rsaCsp = new RSACryptoServiceProvider(); 

        rsaCsp.ImportParameters(rsaParameters); 

        privateKey = rsaCsp;
#endif

        var certificate = new X509Certificate2(bcCertificate.GetEncoded());

        certificate.PrivateKey = privateKey;

        return certificate;
    }
}
```
Then here:
```C#
using (X509Certificate2 certificate = CertificateUtilities.GetCertificateWithPrivateKey(bcCertificate, privateKey))
{
    ...
}
```

Reply:
Thanks! Fixed.